pkg/controller/autoscaling/autoscalinggroup/setup.go

```diff
@@ -179,8 +187,21 @@ func isUpToDate(_ context.Context, obj *svcapitypes.AutoScalingGroup, obs *svcsd
 	sort.Slice(asg.Tags, func(i, j int) bool {
 		return *asg.Tags[i].Key < *asg.Tags[j].Key
 	})
-	if !cmp.Equal(in.Tags, asg.Tags) {
-		return false, "", nil
+	if len(in.Tags) != len(asg.Tags) {
+		return false, "spec.forProvider.tags", nil
+	}
+	for i := range in.Tags {
+		a := in.Tags[i]
+		b := asg.Tags[i]
+		if !ptr.Equal(a.Key, b.Key) {
+			return false, fmt.Sprintf("spec.forProvider.tags[%d].key", i), nil
+		}
+		if !ptr.Equal(a.Value, b.Value) {
+			return false, fmt.Sprintf("spec.forProvider.tags[%d].value", i), nil
+		}
+		if !ptr.Equal(a.PropagateAtLaunch, b.PropagateAtLaunch) {
+			return false, fmt.Sprintf("spec.forProvider.tags[%d].propagateAtLaunch", i), nil
+		}
 	}
```

Comment:
I think it would be simpler to just use `cmp.Diff` with a sort op:

```suggestion
        if diff := compareTags(asg.Tags, in.Tags); diff != "" {
        	return false, "spec.forProvider.tags:" + diff, nil
        }
```

```golang
func compareTags(spec, current []svcapitypes.Tag) string {
	return cmp.Diff(current, spec, cmpopts.SortSlices(func(i, j svcapitypes.Tag) bool {
		return ptr.Deref(i.Key, "") < ptr.Deref(j.Key, "")
	}))
}
```

Reply:
Types of in.Tags and asg.Tags are not same

Reply:
You are right. I did not see that.